proposals/NNNN-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,239 @@
+  
+  nonisolated func test() {
+    _ = value
+    _ = $value
+  }
+}
+```
+
+### Property wrappers with backing reference types
+
+A `let` wrapped property could be useful for reference types like a property wrapper class. Typically property wrappers are written for value types but occasionally a protocol like `NSObject` may require the use of a class. For example:
+```swift
+@propertyWrapper
+class WrapperClass<T> : NSObject {
+  var wrappedValue: T
+
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+class C {
+  @WrapperClass let value: Int
+
+  init(v: Int) {
+    value = v
+  }
+}
+```
+Even though  `value` is declared with a class type property wrapper, it can only be assigned once, preventing any future unintentional changes to the property wrapper class instance in this context. 
+
+
+### SwiftUI 
+
+SwiftUI property wrappers may also benefit from a let declaration. For example, `@ScaledMetric` in its simplest usage can be written with a `let`:
+```swift
+struct ContentView: View {
+  @ScaledMetric let imageSize = 10
+
+  var body: some View {
+    Image(systemName: "heart.fill")
+    .resizable()
+    .frame(width: imageSize, height: imageSize)
+  }
+}
+```
+
+Similarly, other SwiftUI property wrappers could be `let` declared when they do not require more than a single initialization.
+
+## Alternatives considered
+
+### @State and @Binding property wrappers with `nonmutating set`
+
+Currently, SwiftUI offers two property wrappers,  [`@State`](https://developer.apple.com/documentation/swiftui/state/wrappedvalue) and [`@Binding`](https://developer.apple.com/documentation/swiftui/binding/wrappedvalue), that have a `wrappedValue` property with a `nonmutating set`. This allows the backing types to preserve the reference semantics of the `wrappedValue` implementation.
+
+A `var` declared `@State` property currently generates a nonmutating set that reflects `@State`'s reference based storage:
+
+```swift
+@State var weekday: String = "Monday"
+
+private var _weekday: State<String> = State<String>(wrappedValue: "Monday")
+var weekday : String {
+  get { 
+    return _weekday.wrappedValue 
+  }
+  nonmutating set {
+    self._weekday.wrappedValue = value
+  }
+  nonmutating _modify { 
+    yield () 
+  }
+}
+```
+
+Declaring the `@State` property with a `let` will remove the nonmutating setter:
+
+```swift
+@State let weekday: String = "Monday"
+
+private let _weekday: State<String> = State<String>(wrappedValue: "Monday")
+var weekday : String {
+  get { 
+    return _weekday.wrappedValue 
+  }
+}
+```
+
+[There is an argument to be made](https://forums.swift.org/t/pitch-allow-property-wrappers-on-let-declarations/61750/20) in favor of retaining the `nonmutating set` for let declared `@State` and `@Binding` to signify their reference based storage. Afterall, a property wrapper's `wrappedValue` could be assigned via the `_weekday` backing property, even if `weekday` was marked with a `let` instead of a `var`.
```

Comment:
```suggestion
[There is an argument to be made](https://forums.swift.org/t/pitch-allow-property-wrappers-on-let-declarations/61750/20) in favor of retaining the `nonmutating set` here. After all, a property wrapper's `wrappedValue` could be assigned via the `_weekday` backing property, even if `weekday` was marked with a `let` instead of a `var`.
```